parser-typechecker/src/Unison/Codebase.hs

```diff
@@ -347,17 +349,34 @@ typeLookupForDependencies codebase s = do
   foldM go mempty s
   where
     go tl ref@(Reference.DerivedId id) =
-      fmap (tl <>) $
-        getTypeOfTerm codebase ref >>= \case
-          Just typ -> pure $ TypeLookup (Map.singleton ref typ) mempty mempty
-          Nothing ->
-            getTypeDeclaration codebase id >>= \case
-              Just (Left ed) ->
-                pure $ TypeLookup mempty mempty (Map.singleton ref ed)
-              Just (Right dd) ->
-                pure $ TypeLookup mempty (Map.singleton ref dd) mempty
-              Nothing -> pure mempty
+      getTypeOfTerm codebase ref >>= \case
+        Just typ -> pure $ tl <> TypeLookup (Map.singleton ref typ) mempty mempty
+        Nothing ->
+          getTypeDeclaration codebase id >>= \case
+            Just (Left ed) ->
+              pure $ tl <> TypeLookup mempty mempty (Map.singleton ref ed)
+            Just (Right dd) -> do
+              -- We need the transitive dependencies of data decls
+              -- that are scrutinized in a match expression for
+              -- pattern match coverage checking (specifically for
+              -- the inhabitation check). We ensure these are found
+              -- by collecting all type dependencies for all data
+              -- decls.
+              let constructorTypes :: [Type Symbol a]
+                  constructorTypes = snd <$> DD.constructors dd
+
+                  -- All references from constructorTypes that we
+                  -- have not already gathered.
+                  constructorRefs :: Set Reference
+                  constructorRefs = foldl' (\b a -> Set.filter (unseen tl) (Type.dependencies a) <> b) mempty constructorTypes
```

Comment:
Could you use `Unison.DataDeclaration.dependencies` for this? I think it's doing the same thing.